ocaml/xapi/xapi_pool_patch.ml

```diff
@@ -645,8 +645,18 @@ let apply ~__context ~self ~host =
 	        output;
 		
         | Failure(log, exn) ->
-	        debug "error from patch application: %s" log;
-	        raise (Api_errors.Server_error(Api_errors.patch_apply_failed, [log]))
+		debug "error from patch application: %s" log;
+           	let error_string = "Backup files already present - aborting." in
+           	if List.length (Xstringext.String.find_all error_string log) = 0 then
```

Comment:
Searching for this string seems quite brittle - what happens if the patch tool's error message changes slightly?   Is there a better way to detect this failure?